dace/frontend/fortran/fortran_parser.py

```diff
@@ -69,16 +325,29 @@ def __init__(self, ast: ast_components.InternalFortranAst, source: str, use_expl
             ast_internal_classes.Write_Stmt_Node: self.write2sdfg,
             ast_internal_classes.Allocate_Stmt_Node: self.allocate2sdfg,
             ast_internal_classes.Break_Node: self.break2sdfg,
+            ast_internal_classes.Continue_Node: self.continue2sdfg,
+            ast_internal_classes.Derived_Type_Def_Node: self.derivedtypedef2sdfg,
+            ast_internal_classes.Pointer_Assignment_Stmt_Node: self.pointerassignment2sdfg,
+            ast_internal_classes.While_Stmt_Node: self.whilestmt2sdfg,
         }
-        self.use_explicit_cf = use_explicit_cf
 
     def get_dace_type(self, type):
         """
         This function matches the fortran type to the corresponding dace type
         by referencing the ast_utils.fortrantypes2dacetypes dictionary.
         """
         if isinstance(type, str):
-            return ast_utils.fortrantypes2dacetypes[type]
+            if type in ast_utils.fortrantypes2dacetypes:
+                return ast_utils.fortrantypes2dacetypes[type]
+            elif type in self.registered_types:
+                return self.registered_types[type]
+            else:
+                # TODO: This is bandaid.
```

Comment:
why? what is it a band aid for? Please explain in the code

Reply:
Clarified.